src/components/fields/contact-number/types.ts

```diff
@@ -0,0 +1,37 @@
+import { FormInputGroupProps } from "@lifesg/react-design-system/form/types";
+import {
+	IFrontendEngineBaseFieldJsonSchema,
+	TFrontendEngineBaseFieldJsonSchemaKeys,
+} from "../../frontend-engine/types";
+import { IYupValidationRule } from "../../frontend-engine/yup/types";
+import { InternationalCallingCodeMap } from "./data";
+
+export type TCountry = keyof typeof InternationalCallingCodeMap;
+
+interface IContactNumberProps extends FormInputGroupProps<string, unknown> {
+	country?: TCountry;
+	allowInternationalNumbers?: boolean;
+	enableSearch?: boolean;
+}
+
+interface IContactNumberValidationRules extends IYupValidationRule {
+	isSingaporeNumber?: {
```

Comment:
aside from accepting as boolean, also need to make sure either `homeNumber` or `mobileNumber` is defined, so it needs to be a union type
```
| true
| {
		isHomeNumber: true;
		isMobileNumber?: boolean;
	}
| {
		isHomeNumber?: boolean;
		isMobileNumber: true;
	};
```